File: DSA/vaishnavi_joshi_590016947/Day71/Q2.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String s = sc.nextLine().replaceAll(" ", "");

        int L = s.length();

        int rows = (int) Math.floor(Math.sqrt(L));
        int cols = (int) Math.ceil(Math.sqrt(L));

        if (rows * cols < L) {
            rows++;
        }

        StringBuilder ans = new StringBuilder();

        for (int col = 0; col < cols; col++) {
            for (int row = 0; row < rows; row++) {
                int index = row * cols + col;

                if (index < L) {
                    ans.append(s.charAt(index));
                }
            }

            if (col < cols - 1) {
                ans.append(" ");
            }
        }

        System.out.println(ans);
        sc.close();
    }
}

File: DSA/VivaswanSingh_590023022/Day_71/q2.c
#include <stdio.h>
#include <string.h>
#include <math.h>

void encryption(char *s)
{
    char str[10000];
    int len = 0;

    // Remove spaces
    for (int i = 0; s[i] != '\0'; i++)
    {
        if (s[i] != ' ')
            str[len++] = s[i];
    }

    str[len] = '\0';

    int rows = (int)sqrt(len);
    int cols = rows;

    if (rows * cols < len)
        cols++;

    if (rows * cols < len)
        rows++;

    for (int col = 0; col < cols; col++)
    {
        int printed = 0;

        for (int row = 0; row < rows; row++)
        {
            int index = row * cols + col;

            if (index < len)
            {
                if (printed)
                    printf("%c", str[index]);

                else
                {
                    printf("%c", str[index]);
                    printed = 1;
                }
            }
        }

        if (col != cols - 1)
            printf(" ");
    }

    printf("\n");
}

int main()
{
    char s[10000];

    fgets(s, sizeof(s), stdin);

    encryption(s);

    return 0;
}